source/RSRCFilter.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Raheem Idowu <abdurraheemidowu@gmail.com>
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+#include "RSRCFilter.h"
+
+#include <MimeType.h>
+#include <NodeInfo.h>
+#include <string.h>
+
+#include <compat/sys/stat.h>
+
+static const char* supported_types[] = {
+	"application/x-vnd.be-elfexecutable",
+	NULL
+};
+
+bool
+RSRCFilter::Filter(const entry_ref* ref, BNode* node, struct stat_beos* st,
+	const char* filetype)
+{
+	BEntry entry(ref, true);
+	if (entry.InitCheck() != B_OK)
+		return false;
+	if (entry.IsDirectory())
+		return true;
+
+	entry_ref entryRef;
+	if (entry.GetRef(&entryRef) != B_OK)
+		return false;
+
+	//Check if ref is a symlink
+	bool isLink = S_ISLNK(st->st_mode);
+	if (isLink) {
+		BNodeInfo * nodeInfo = new BNodeInfo(node);
```

Comment:
`node` is the file node of symlink `ref`, but you need the file node of `entry` or `entryRef`.